server/container_portforward.go

```diff
@@ -56,5 +59,6 @@ func (s StreamService) PortForward(podSandboxID string, port int32, stream io.Re
 	}
 
 	emptyStreamOnError = false
-	return s.runtimeServer.Runtime().PortForwardContainer(c, port, stream)
+
+	return s.runtimeServer.Runtime().PortForwardContainer(ctx, c, sb.NetNsPath(), port, stream)
```

Comment:
NetNsPath() has the ability to return "", which I would deem an error here

Reply:
The question is where we catch that. Is the `sb.NetNsPath() == ""` if we're running on host network? If yes, should we do the same like there: https://github.com/containerd/cri/blob/64aa9da76fc0ab333119f455f3b292244c1fae8c/pkg/server/sandbox_portforward_unix.go#L58-L60 ?

Reply:
not quite. 
NetNsPath() returns "" iff we're `manage_ns_lifecycle` = false and the infra container PID is not initialized. Otherwise, we send `/proc/$pid/ns/net`, or `/var/run/netns/$id`
thus, empty NetNsPath is an error here (we somehow started port forwarding in a container without a pod infra container initialized)

Reply:
Thanks for the clarification, I added an error check